### PR TITLE
Update queues in Runtime Classes KHRGA-58

### PR DESCRIPTION
**Pull request by imvucic**

### CODE REVIEW

File: source/iface/queue.rst [hunk 1]
```diff
@@ -116,23 +128,30 @@ invalid_object_error
...
+
 ``get_context``
 ===============
 
 ::
 
   context get_context() const;
 
-Returns context associated with queue.
+Returns the SYCL queue’s context. Reports errors using SYCL exception classes.
```
Comment (gmlueck) [inline source/iface/queue.rst:141]: This wording about reporting errors seems confusing to me.  I proposed removing it from the spec (KhronosGroup/SYCL-Docs#471).  I suggest removing it from the reference guide too.

File: source/iface/queue.rst [hunk 2]
```diff
@@ -116,23 +128,30 @@ invalid_object_error
...
-===============
+==============
 
 ::
 
   device get_device() const;
 
-Returns device associated with queue.
+Returns the SYCL device the queue is associated with. Reports errors using
+SYCL exception classes. The value returned must be equal to that returned
```
Comment (gmlueck) [inline source/iface/queue.rst:153]: Same here.

File: source/iface/queue.rst [hunk 3]
```diff
@@ -104,10 +112,14 @@ from asynchronous host tasks, construct the queue with an
```
[the remark this answers is not in the document]
Comment (imvucic) [reply source/iface/queue.rst:110]: Fixed with commit: https://github.com/KhronosGroup/SYCL_Reference/pull/74/commits/82e7c5b4823c084ecf102d61b6505693ffc0e63d

File: source/iface/queue.rst [hunk 4]
```diff
@@ -141,25 +160,63 @@ Returns device associated with queue.
...
+``get_info``
+============
+
+::
+
+  template <typename Param> typename Param::return_type get_info() const;
+
+Queries this SYCL ``queue`` for information requested by the template
+parameter ``Param``. The type alias ``Param::return_type`` must be defined
+in accordance with the info parameters in |SYCL_SPEC_QUEUE_INFO_DESC|
```
[the remark this answers is not in the document]
Comment (imvucic) [reply source/iface/queue.rst]: Fixed with commit: https://github.com/KhronosGroup/SYCL_Reference/pull/74/commits/82e7c5b4823c084ecf102d61b6505693ffc0e63d

File: source/iface/queue.rst [hunk 5]
```diff
@@ -168,10 +225,30 @@ queue_ for details.
...
-  event submit(T cgf, const queue &secondaryQueue);
+  template <typename T> event submit(T cgf);
+
+  template <typename T> event submit(T cgf, queue& secondaryQueue);
+
+Submit a command group function object to the queue, in order to be scheduled
+for execution on the device. On a kernel error, this command group function
+object is then scheduled for execution on the secondary queue. Returns
+an event, which corresponds to the queue the command group function object
+is being enqueued on.
```
[the remark this answers is not in the document]
Comment (imvucic) [reply source/iface/queue.rst:234]: Fixed with commit: https://github.com/KhronosGroup/SYCL_Reference/pull/74/commits/82e7c5b4823c084ecf102d61b6505693ffc0e63d

File: source/iface/queue.rst [hunk 6]
```diff
@@ -168,10 +225,30 @@ queue_ for details.
...
+  typename Param::return_type get_backend_info() const;
+
+Queries this SYCL ``queue`` for SYCL backend-specific information requested by
+the template parameter ``Param``. The type alias ``Param::return_type``
+must defined in accordance with the SYCL backend specification. Must throw
+an ``exception`` with the ``errc::backend_mismatch`` error code if the SYCL
+backend that corresponds with ``Param`` is different from the SYCL backend
+that is associated with this ``queue``.
 
 .. rubric:: Template parameters
```
[the remark this answers is not in the document]
Comment (imvucic) [reply source/iface/queue.rst:236]: Fixed with commit: https://github.com/KhronosGroup/SYCL_Reference/pull/74/commits/82e7c5b4823c084ecf102d61b6505693ffc0e63d

File: source/iface/queue.rst [hunk 7]
```diff
@@ -207,55 +284,28 @@ if an error occurs executing on the primary queue.
...
-======================
-
-::
-
-  void throw_asynchronous();
-
-Passes any asynchronous errors to handler provided in
-:ref:`queue-constructors`.
+The full list of queue shortcuts is
+defined in |SYCL_SPEC_QUEUE_SHOURTCUT_FUNC|.
```
[the remark this answers is not in the document]
Comment (imvucic) [reply source/iface/queue.rst]: Fixed with commit: https://github.com/KhronosGroup/SYCL_Reference/pull/74/commits/82e7c5b4823c084ecf102d61b6505693ffc0e63d

File: source/iface/queue.rst (hunk 1 above)
Comment (imvucic) [reply source/iface/queue.rst]: Fixed with commit: https://github.com/KhronosGroup/SYCL_Reference/pull/74/commits/82e7c5b4823c084ecf102d61b6505693ffc0e63d

File: source/iface/queue.rst (hunk 2 above)
Comment (imvucic) [reply source/iface/queue.rst]: Fixed with commit: https://github.com/KhronosGroup/SYCL_Reference/pull/74/commits/82e7c5b4823c084ecf102d61b6505693ffc0e63d

File: source/iface/queue.rst [hunk 8]
```diff
@@ -168,10 +222,16 @@ queue_ for details.
...
+  template <typename T> event submit(T cgf, queue& secondaryQueue);
+
+Submit a command group function object to the queue, in order to be scheduled
+for execution on the device. In the context of the second overload,
+if a kernel error occurs, this command group function object is then
+scheduled for execution on the secondary queue. Returns an event,
+which corresponds to the queue the command group function object
+is being enqueued on.
 
 .. rubric:: Template parameters
```
Comment (gmlueck) [inline source/iface/queue.rst:236]: This section "Template parameters" doesn't have any information.  Personally, I don't think it's necessary, so I think we could just remove this section.  However, if you want to keep it, you should add some description about the `T` template parameter.

File: source/iface/queue.rst [hunk 9]
```diff
@@ -207,55 +267,48 @@ if an error occurs executing on the primary queue.
...
-
-Passes any asynchronous errors to handler provided in
-:ref:`queue-constructors`.
+Queue shortcut functions are part of the queue class and simplify command
+group creation. They generate a command group with a single command,
+invoke a corresponding member function in the handler, and submit the group.
+The key distinction is that handler member functions return void,
+while queue shortcuts return an event object representing the submitted
+command group. You can specify a list of events to wait on, just like
+using handler::depends_on for the implicit command group.
```
Comment (gmlueck) [inline source/iface/queue.rst:296]: Should `handler::depends_on` be in code font?

File: source/iface/queue.rst [hunk 10]
```diff
@@ -20,70 +20,78 @@ Queues
...
-:ref:`device` as needed.
+The default constructor of the SYCL ``queue`` class will construct
+a queue based on the SYCL ``device`` returned from the
+``default_selector_v`` (see |SYCL_SPEC_DEVICE_SELECTORS|).
+All other constructors construct a queue as determined by the
+parameters provided. All constructors will implicitly construct a SYCL
+``platform``, ``device`` and ``context`` in order to
+facilitate the construction of the queue.
+
+Each constructor takes as the last parameter an optional SYCL ``property_lis``
```
[the remark this answers is not in the document]
Comment (imvucic) [reply source/iface/queue.rst:88]: Fixed with commit : https://github.com/KhronosGroup/SYCL_Reference/pull/74/commits/f4794d57b42f7da711069e1e729304bedfedd85b

[the remark this answers is not in the document]
Comment (gmlueck) [reply source/iface/queue.rst:88]: It looks like this was not fixed in that commit.

File: source/iface/queue.rst [hunk 11]
```diff
@@ -20,70 +20,78 @@ Queues
...
+to provide properties to the SYCL ``queue``.
 
 The SYCL runtime may execute tasks asynchronously. To catch exceptions
 from asynchronous host tasks, construct the queue with an
-``asyncHandler`` and call wait_and_throw_.
+``asyncHandler`` and call wait_ or wait_and_throw_.
```
[the remark this answers is not in the document]
Comment (mkinsner) [reply source/iface/queue.rst:93]: The methods to process asynchronous errors are `wait_and_throw` and `throw_asynchrous`, as detailed at https://registry.khronos.org/SYCL/specs/sycl-2020/html/sycl-2020.html#subsubsec:exception.async

So this text should be updated to `queue::wait_and_throw()`, `queue::throw_asynchronous()`, or `event::wait_and_throw()`

File: source/iface/queue.rst (hunk 10 above)
[the remark this answers is not in the document]
Comment (mkinsner) [reply source/iface/queue.rst:88]: Agreed. I still see `property_lis` when I render the latest commit

File: source/iface/queue.rst [hunk 12]
```diff
@@ -516,10 +578,30 @@ SYCL queue provides in-order semantics.
...
+associated host API calls have returned (so can’t be thrown
+as exceptions by the API call), and that are handled by an
+``async_handler`` through which the errors are reported. Handling
+of asynchronous errors from a ``queue`` occurs at specific times, as described
+by |SYCL_SPEC_ERROR_HANDLING|.
+
+
 =========
 Example 1
 =========
```
[the remark this answers is not in the document]
Comment (mkinsner) [reply source/iface/queue.rst:603]: I agree that we should mention these preferred forms now, as we see these examples go by.  That will save rework later.

File: source/iface/queue.rst (hunk 10 above)
[the remark this answers is not in the document]
Comment (imvucic) [reply source/iface/queue.rst]: Sorry I was thinking only about new line deletion...The list fixed with : https://github.com/KhronosGroup/SYCL_Reference/pull/74/commits/cc1c097adb842538ca70f199a6e6d54b31528016

File: source/iface/queue.rst (hunk 11 above)
[the remark this answers is not in the document]
Comment (imvucic) [reply source/iface/queue.rst]: Dene with : https://github.com/KhronosGroup/SYCL_Reference/pull/74/commits/cc1c097adb842538ca70f199a6e6d54b31528016

File: source/iface/queue.rst (hunk 12 above)
[the remark this answers is not in the document]
Comment (imvucic) [reply source/iface/queue.rst:604]: Following CTAD version with:
https://github.com/KhronosGroup/SYCL_Reference/pull/74/commits/cc1c097adb842538ca70f199a6e6d54b31528016

**Review by gmlueck — APPROVED** [review]

**Review by mkinsner — APPROVED** [review]

**Pull request merged**